pkg/apply/cache/resource_cache.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package cache
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+// ResourceCache stores unstructured resource objects in memory
+type ResourceCache interface {
```

Comment:
Are implementations supposed to be thread safe? I can see that it's being written to from one goroutine and read from another one. The `ResourceCacheMap` implementation is not thread safe and there appears to be no synchronization around it.

Reply:
TIL the builtin Go map isn't thread safe. I'll go add some locks.

Reply:
Fixed.